Common/include/linear_algebra/CSysMatrix.hpp

```diff
@@ -630,7 +594,7 @@ class CSysMatrix {
    * \param[in] block_i - Diagonal index.
    * \param[in] val_block - Block to add to the diagonal of the matrix.
    */
-  template<class OtherType>
+  template<class OtherType, int Sign = 1>
```

Comment:
I think this default template arg might be better as an input arg to the function.. I could see this causing confusion. Same with keeping the name AddBlock2Diag and using for subtraction. I like the reuse you introduced here, but can you make it a little clearer?